cmsl1t/plotting/resolution_vs_X.py

```diff
@@ -0,0 +1,84 @@
+from rootpy import asrootpy
+
+
+class ResolutionVsXPlot(BasePlotter):
+    def __init__(self, resolution_type, online_name, offline_name, versus_name):
+        name = ["resolution_vs_" + versus_name, online_name, offline_name]
+        super(ResolutionVsXPlot, self).__init__("__".join(name))
+        self.online_name = online_name
+        self.offline_name = offline_name
+        self.versus_name = versus_name
+        self.resolution_method = get_resolution_function(resolution_type)
+
+    def create_histograms(self,
+                          online_title, offline_title, versus_title,
+                          pileup_bins, res_n_bins, res_low, res_high,
+                          vs_n_bins, vs_low, vs_high,
+                          ):
+        """ This is not in an init function so that we can by-pass this in the
+        case where we reload things from disk """
+        self.online_title = online_title
+        self.offline_title = offline_title
+        self.versus_title = versus_title
+        self.pileup_bins = bn.Sorted(pileup_bins, "pileup",
+                                     use_everything_bin=True)
+
+        name = ["resolution_vs", self.versus_name, self.online_name, self.offline_name, "pu_{pileup}"]
+        name = "__".join(name)
+        title = " ".join(["Resolution (", self.online_name, "vs.", self.offline_name,
+                          ") against", self.versus_name, "in PU bin: {pileup}"])
+        title = ";".join([title, self.offline_title, self.online_title])
+        self.plots = HistogramCollection([self.pileup_bins],
+                                         "Hist2D", vs_n_bins, vs_low, vs_high,
+                                         res_n_bins, res_low, res_high,
+                                         name=name, title=title)
+        self.filename_format = name
+
+    def fill(self, pileup, versus, offline, online):
+        difference = self.resolution_method(online, offline)
+        self.plots[pileup].fill(versus, difference)
+
+    def draw(self, with_fits=True):
+        for (pileup, ), hist in self.plots.flat_items_all():
+            self.__do_draw(pileup, hist)
+            self.__do_draw(pileup, asrootpy(hist.ProfileX()), "_profile")
+
+    def __do_draw(self, pileup, hist, suffix=""):
+        with preserve_current_style():
+            # Draw each efficiency (with fit)
+            ytitle = self.resolution_method.label.format(on=self.online_title, off=self.offline_title)
+            canvas = draw2D(hist, draw_args={"xtitle": self.versus_title, "ytitle": ytitle})
+
+            # Add labels
+            label_canvas()
+
+            # Save canvas to file
+            name = self.filename_format.format(pileup=pileup)
+            self.save_canvas(canvas, name + suffix)
+
+    def _is_consistent(self, new):
+        """
+        Check the two plotters are the consistent, so same binning and same axis names
+        """
+        return (self.pileup_bins.bins == new.pileup_bins.bins) and \
```

Comment:
tempted to modify this for readability:
```python
attrsToCompare =['resolution_method', 'versus_name', 'online_name', 'offline_name']
results = [getattr(self, attr) == getattr(new, attr) for atttr in attrsToCompare]
results.append(self.pileup_bins.bins == new.pileup_bins.bins)
return all(results)
```

On second thought, maybe better to
```python
return all ([
    self.pileup_bins.bins == new.pileup_bins.bins,
    self.resolution_method == new.resolution_method,
    self.versus_name == new.versus_name,
    self.online_name == new.online_name,
    self.offline_name == new.offline_name
])
```
(fewer 'ands')

Reply:
I prefer using `all` the most.  The first option feels actually less clear.  An alternative, for the future maybe, is to have each plotter specify which attributes need to be checked and have the base class actually implement the checking of these.